packages/nango-yaml/lib/parser.ts

```diff
@@ -168,6 +172,10 @@ export abstract class NangoYamlParser {
                 }
                 if (action.endpoint) {
                     for (const [verb, endpoint] of Object.entries(action.endpoint)) {
+                        if (!endpoint) {
+                            continue; // TS pleasing
+                        }
```

Comment:
why do we need to allow undefined value in 
```
export type NangoSyncEndpoint = {
    [key in HTTP_VERB]?: string;
};
```
An endpoint without a path should be valid and representable

Reply:
because zod optional() is setting `VERB?: string | undefined`

Reply:
I copy pasted the wrong line. I don't mean the undefined key I mean the undefined value
```
    [key in HTTP_VERB]?: string | undefined;
```

Reply:
yep, I set that because otherwise the type is not compatible with zod type. There might be a way to improve the zod type I just didn't found how

Reply:
what about something like 
```
            endpoints: z
                .array(
                    z.record(
                       z.union([z.literal('GET'), z.literal('PUT'), z.literal('POST'), ...]), 
                       z.string()
                    )
                )
                .optional(),

```

Reply:
indeed, it works I find it way less clear to understand tho and doesn't enforce that any is set.
I think the only viable alternative is to have 
`z.array(z.union([z.object({ GET: z.string }), z.object({ POST: z.string() }) ...]});`

Reply:
harder to read schema/types vs not being able to represent state that are impossible: for me the choice would be easy ;)